src/frequenz/sdk/timeseries/logical_meter/_formula_generators/_battery_power_formula.py

```diff
@@ -0,0 +1,69 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""Formula generator from component graph for Grid Power."""
+from .....sdk import microgrid
+from ....microgrid.component import ComponentCategory, ComponentMetricId
+from .._formula_engine import FormulaEngine
+from .._resampled_formula_builder import ResampledFormulaBuilder
+from ._formula_generator import FormulaGenerator
+
+
+class BatteryPowerFormula(FormulaGenerator):
+    """Creates a formula engine from the component graph for calculating grid power."""
+
+    async def generate(
+        self,
+    ) -> FormulaEngine:
+        """Make a formula for the cumulative AC battery power of a microgrid.
+
+        The calculation is performed by adding the Active Powers of all the inverters
+        that are attached to batteries.
+
+        If there's no data coming from an inverter, that inverter's power will be
+        treated as 0.
+
+        Returns:
+            A formula engine that will calculate cumulative battery power values.
+
+        Raises:
+            RuntimeError: if there are no batteries in the component graph, or if they
+                don't have an inverter as a predecessor.
+        """
+        component_graph = microgrid.get().component_graph
+        batteries = list(
+            comp
+            for comp in component_graph.components()
+            if comp.category == ComponentCategory.BATTERY
+        )
+
+        if not batteries:
+            raise RuntimeError("Unable to find any batteries in the component graph.")
+
+        battery_inverters = list(
+            next(iter(component_graph.predecessors(bat.component_id)))
+            for bat in batteries
+        )
+
+        if len(batteries) != len(battery_inverters):
+            raise RuntimeError("Can't find inverters for all batteries.")
+
```

Comment:
Why not to look for ComponentCategory.INVERTER?
And then iterate once over returned list to find inverters with ComponentType.BATTERY?

Reply:
They are both the same right?  Do you think it is better to start from the inverters for some reason?

Reply:
I am telling that you don't need batteries, here. :) 
```
inverters = list(
            comp
            for comp in component_graph.components()
            if comp.category == ComponentCategory.INVERTER
        )
battery_inverters = list(inverter for inverter in inverters if inverter.component_type == InverterType.BATTERY)
for idx, comp in enumerate(battery_inverters):
   ....
```

Reply:
done